Feature build static gcc

sarfata opened the pull request:
Build GCC without external dependencies.

sarfata committed:
Build GCC statically (remove dependency to libmpc, etc)

 - Build libmpc, libmpfr and libgmp in build/libs
   The trick here is that we build them with --disable-shared
 - Compile gcc and use those libs. Because the shared are not available
   it will automatically compile statically

That's it! Was pretty easy ;)

martijnthe commented:
Looks good to me. @bradtgmurray ?

bradtgmurray commented on Makefile:
```diff
@@ -275,7 +277,9 @@ cross-gcc: cross-binutils cross-gcc-first cross-newlib gcc-$(GCC_VERSION)-$(CS_B
 	--with-sysroot="$(PREFIX)/$(TARGET)"				\
 	--with-build-time-tools="$(PREFIX)/$(TARGET)/bin"		\
-	--enable-extra-sgxxlite-multilibs $(CS_SPECS) && \
+	--with-gmp=$(PREFIX) --with-mpfr=$(PREFIX) --with-mpc=$(PREFIX) \
```
with-gmp and friends are here twice?

sarfata replied:
That's a mistake that careful diff examination should have caught. Shame on me :confounded:

bradtgmurray commented on Makefile:
```diff
@@ -241,7 +242,7 @@ CS_SPECS='--with-specs=%{save-temps: -fverbose-asm}			\
 -fremove-local-statics}}}'
 
-cross-gcc-first: cross-binutils gcc-$(GCC_VERSION)-$(CS_BASE) multilibbash 
+cross-gcc-first: gmp mpfr mpc cross-binutils gcc-$(GCC_VERSION)-$(CS_BASE) multilibbash
```
As far as I know the changes to this target are unnecessary. gcc-first is only built as a build artifact, it's a compiler that's only built to build newlib and then it's used to rebuild gcc itself in the cross-gcc target. End users will only use the final one. However, it doesn't hurt...

sarfata replied:
You need this to be able to build without the libs installed on your machine. I had removed them to make sure that it was built with few external deps. I think it makes sense to keep that in, it also simplifies the requirements if someone wants to rebuild the tools.

bradtgmurray commented:
:+1:

sarfata commented:
Not merging yet because I have a problem building this on ubuntu 13.04 ... I will try to get one single script that builds everywhere before I merge.

sarfata committed:
Removed one extra line of --with-gmp-and-friend

Meiguro committed:
Add incremental build support

Meiguro committed:
Add incremental tar extraction

Meiguro committed:
Add incremental configuration

Meiguro committed:
Remove cross-newlib as a install-tools dep

cross-newlib is a cross-gcc dep so it will still be run. Pulling
cross-newlib up to a install-tools dep causes gross-gcc to recompile.

Meiguro committed:
Add --enable-static configure flag for mpfr and mpc

Meiguro committed:
Set install-deps as phony

Meiguro committed:
Ignore libs directory

Meiguro committed:
Add -Wno-error=return-type CFLAG for building gdb with clang

Meiguro committed:
Merge pull request #3 from pebble/feature-incremental-build

Feature incremental build

sarfata commented:
Closing this because this was included in the incremental build PR of @Meiguro which is now merged.
